src/app/components/shared/header/header.component.ts

```diff
@@ -2,18 +2,66 @@ import { Component, OnInit } from '@angular/core';
 import { MenuItem } from 'primeng/api';
 import { OverlayService } from '../../../service';
 import { SingInComponent } from '../../sing-in/sing-in.component';
+import { SingInService } from '../../../service';
+import { PopUpComponent } from '../../popup/popup.component';
+import { Router } from '@angular/router';
 
 @Component({
   selector: 'header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
 })
+
 export class HeaderComponent implements OnInit {
   items: MenuItem[] = [];
+  isLoggedIn: boolean = false;
 
-  constructor(private dialog: OverlayService) {}
+  singInBtn: MenuItem = {
+    label: 'Sing in',
+    icon: 'pi pi-fw pi-sign-in',
+    command: () => {
+      this.dialog.open(SingInComponent);
+    },
+  }
+
+  logOutBtn: MenuItem = {
+    label: 'Log out',
+    icon: 'pi pi-fw pi-sign-out',
+    command: () => {
+      const dialogRef = this.dialog.open(PopUpComponent, {
+        data: {
+          title: 'Do you want to log out?',
+          btnLabel1: 'Yes',
+          btnLabel2: 'No',
+        },
+        backdropClickClose: true,
+      });
+      dialogRef.afterClosed().subscribe((result) => {
+        if (result) {
+          this.singInService.login();
```

Comment:
Using 'login' method is unexpected here as we are going to logout. It may be better to not allow call the 'login' method without parameter

Reply:
Fixed